src/bin/server.rs

```diff
@@ -38,26 +39,71 @@ fn main() {
     cfg.set_str("user.name", "bors").unwrap();
     cfg.set_str("user.email", "bors@rust-lang.org").unwrap();
 
+    let api_protocol = String::from("https");
+    let mirror = env::var("MIRROR").is_ok();
+
     let heroku = env::var("HEROKU").is_ok();
     let cargo_env = if heroku {
         cargo_registry::Env::Production
     } else {
         cargo_registry::Env::Development
     };
+
+    let uploader = match (cargo_env, mirror) {
+        (cargo_registry::Env::Production, false) => {
```

Comment:
Do you think this code would be a bit more self documenting if an enum with descriptively named variants were used instead of a boolean?